### PR TITLE
Extend `yamllint` hook

### PR DESCRIPTION
**Pull request by totoroot**
I added a couple of options to the `yamllint` hook, as so far it was only possible to pass a path to the configuration file.
In some comments, I also reasoned why some options weren't added as they do not make sense for a pre-commit hook.

Similar to how I did it for the `typos` hook, there is now also an option to pass a multiline string as configuration to `yamllint`. Additionally `yamllint` allows for a serialized configuration, but the multiline string configuration option is the preferred way. Of course just providing a path to a configuration file remains possible.

The behaviour of the hook itself mostly stayed the same, although the `relaxed` option was removed in favour of a more generalized `preset` option, that allows us to easily add presets, should more ever be added to `yamllint`.

A pre-commit hook only makes sense, if it prevents you from committing if errors were found, so `--strict` should definitely be the default.

### Breaking changes:
- `settings.yamllint.relaxed = true;` is now `settings.yamllint.preset = "relaxed";`
- `settings.yamllint.strict` was added and defaults to `true` :arrow_right: strict error checking for hook

### Additional changes:
For some hooks I recently worked on, I noticed that I was shadowing the top-level attribute `config`, which is never a good idea, so I changed the multiline configuration options to `configuration`.
Affected hooks are `typos` and `vale`. The options were only added recently.

### CODE REVIEW

**Comment by domenkozar** [comment]
Sorry for the mess, we revamed module structure in https://github.com/cachix/pre-commit-hooks.nix/pull/397, could you port these changes over?

File: modules/hooks.nix [hunk 1]
```diff
@@ -1340,7 +1340,7 @@ in
                 default = "auto";
               };
 
-            config =
+            configuration =
```
Comment (domenkozar) [inline modules/hooks.nix:1343]: we should warn about this option rename and deprecate `relaxed`

Comment (totoroot) [reply modules/hooks.nix:1343]: Sure. How do I create a warning?

Comment (totoroot) [reply modules/hooks.nix:1351]: I tried adding them in 2602c35e7dca1e4d4c32b8f4b308e555948069ba but it fails with the following:
```nix
error: evaluation aborted with the following error message: 'cannot find attribute `hooks.yamllint.settings.relaxed''
```

I'm sure I'm missing something but from what I've seen in examples I used the modules correctly. Any help is appreciated.

Comment (totoroot) [reply modules/hooks.nix:1351]: @domenkozar

**Comment by domenkozar** [comment]
error: evaluation aborted with the following error message: 'cannot find attribute `hooks.yamllint.settings.relaxed''

File: modules/hooks.nix (hunk 1 above)
Comment (totoroot) [reply modules/hooks.nix:1382]: I fixed all warnings.

**Comment by domenkozar** [comment]
Ci is failing

**Pull request merged**